traffic_ops/traffic_ops_golang/servers.go

```diff
@@ -163,3 +167,266 @@ JOIN status st ON s.status = st.id
 JOIN type t ON s.type = t.id`
 	return query
 }
+
+func assignDsesToServerHandler(db *sqlx.DB) AuthRegexHandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request, params PathParams, username string, privLevel int) {
+		handleErr := func(err error, status int) {
+			log.Errorf("%v %v\n", r.RemoteAddr, err)
+			errBytes, jsonErr := json.Marshal(CreateErrorAlerts(err))
+			if jsonErr != nil {
+				log.Errorf("failed to marshal error: %s\n", jsonErr)
+				w.WriteHeader(http.StatusInternalServerError)
+				fmt.Fprintf(w,http.StatusText(http.StatusInternalServerError))
+				return
+			}
+			w.WriteHeader(status)
+			fmt.Fprintf(w,"%s", errBytes)
+		}
+		decoder := json.NewDecoder(r.Body)
+
+		var dsList []int
+
+		err := decoder.Decode(&dsList)
+		if err != nil {
+			handleErr(err,http.StatusInternalServerError)
+			return
+		}
+
+		q := r.URL.Query()
+
+		replaceQueryParameter := q["replace"][0]
+		replace, err := strconv.ParseBool(replaceQueryParameter) //accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False. for replace url parameter documentation
+		if err != nil {
+			handleErr(err, http.StatusBadRequest)
+			return
+		}
+
+		serverPathParameter := params["server"]
+		server, err := strconv.Atoi(serverPathParameter)
+		if err != nil {
+			handleErr(err, http.StatusBadRequest)
+			return
+		}
+
+		assignedDSes, err := assignDeliveryServicesToServer(server, dsList, replace, db)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		resp := struct {
+			Response []int `json:"response"`
+			Alerts
+		}{assignedDSes, CreateAlerts(SuccessLevel,"successfully assigned dses to server")}
+		respBts, err := json.Marshal(resp)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", respBts)
+	}
+}
+
+type Parameter struct{
+	ConfigFile string
+	Name string
+	Value string
+}
+
+func assignDeliveryServicesToServer(server int, dses []int, replace bool, db *sqlx.DB) ([]int, error) {
+	tx, err := db.Beginx()
+	defer func() {
+		if tx != nil {
+			if err != nil {
+				tx.Rollback()
+				return
+			}
+			tx.Commit()
+		}
+	}()
+
+	DBError := fmt.Errorf("database access error")
```

Comment:
I'd vote putting this in a const/var public to the package, so callers can easily check `if err == apipkg.DBErr {`

Reply:
This error wont be being checked for, it is just so we don't send actual db errors to client.

Reply:
other endpoints should use it as well,  and may want to key off that error for logging.